### PR TITLE
Create a minimal Makefile snippet for Spack builds based on make.ubuntu and enable deterministic builds

### PR DESCRIPTION
**Pull request by harshula**

### CODE REVIEW

File: util/make_dir/make.spack [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+AR          = ar
+ARFLAGS     = -rvD
+#
+# CPP keys and compiler options
+#
+CPPDEF    = -Duse_netCDF -Duse_comm_$(CHAN) -D__VERBOSE -DTREAT_OVERLAY
+#
+#
+
+F90FLAGS_1  = -Wall -fallow-argument-mismatch
```
Comment (aidanheerdegen) [inline util/make_dir/make.spack:35]: I can't see where fortran flags can be passed through to the build from spack.

File: util/make_dir/make.spack [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+f90FLAGS_1  = $(F90FLAGS_1)
+FFLAGS_1    = $(F90FLAGS_1)
+fFLAGS_1    = $(F90FLAGS_1)
+CCFLAGS_1   =
+LDFLAGS     =
+MCT_FCFLAGS = -Wall -fallow-argument-mismatch
```
Comment (aidanheerdegen) [inline util/make_dir/make.spack:41]: Same as above re build flags from spack.

File: util/make_dir/make.spack (hunk 1 above)
Comment (harshula) [reply util/make_dir/make.spack:35]: We should be able to do it the same way as F90 and CC by overriding the variable from the Spack package file. e.g.
`            build("-f", "TopMakefileOasis3", "F90=mpif90", "CC=gcc")`

File: util/make_dir/make.spack (hunk 2 above)
Comment (harshula) [reply util/make_dir/make.spack:41]: We should be able to do it the same way as F90 and CC by overriding the variable from the Spack package file. e.g.
`            build("-f", "TopMakefileOasis3", "F90=mpif90", "CC=gcc")`

File: util/make_dir/make.spack (hunk 1 above)
Comment (aidanheerdegen) [reply util/make_dir/make.spack:35]: Right, so we can redefine `F90FLAGS_1`, but we can't add initialise a fortran flags variable that gets added to in the make snippet? Or put the other way, we can't add flags to the defaults in the snippet?

Comment (harshula) [reply util/make_dir/make.spack:35]: I think we can redefine any of the variables. However, I don't think we can append to them.

**Review by aidanheerdegen — APPROVED** [review]

**Comment by harshula** [comment]
The approach of adding the Makefile snippet to the oasis3-mct repository ended up being a waste of time. It's really easy to create the Makefile snippet from Spack and provides far greater flexibility.

**Pull request merged**